core/base/src/TDirectory.cxx

```diff
@@ -213,7 +213,10 @@ void TDirectory::Append(TObject *obj, Bool_t replace /* = kFALSE */)
    }
 
    fList->Add(obj);
-   obj->SetBit(kMustCleanup);
+   // A priori TDirectory will not be 'shared owned' or referenced.
+   // If they are, let's really on the user to update the bit.
```

Comment:
```suggestion
   // A priori, a `TDirectory` object is assumed to not have shared ownership.
   // If it is, let's rely on the user to update the bit.
```

Reply:
> I ignore why this was originally making the cleanup slow 

The 'kMustCleanup' flags tells the `TObject` destructor to inform the `gROOT->GetListOfCleanups()` (via a call to `RecursiveRemove` that the object is being deleted.   `TFile` are on that list by default and the call to `RecursiveRemove` will (of course) recursively call `RecursiveRemove`.  The `TDirectory` then have to lookup in their list if the deleted object is there.  So in this case there is O(N^2) behavior/cost to do ... exactly nothing ...